core/src/main/java/org/mobilitydata/gtfsvalidator/notice/NoticeContainer.java

```diff
@@ -152,4 +173,195 @@ public String exportSystemErrors(boolean isPretty) {
+      clazz = noticeClass.load();
+      if (ValidationNotice.class.isAssignableFrom(clazz)) {
+        // do not iterate over ValidationNotice.java which is the base class for all validation
+        // notices
+        if (clazz.equals(ValidationNotice.class)){
+          continue;
+        }
+        toReturn.add(
+            noticeClass.getSimpleName(),
+            extractNoticeProperties(clazz)
+        );
+      }
+    }
+    return toReturn;
+  }
+
+  /**
+   * Extract information from notices defined in the main module.
+   *
+   * @param validatorPackageName the name of the package that contains the {@code GtfsValidator}s
+   * @return the {@code com.google.gson.JsonObject} that contains the {@code ValidationNotice} class
+   * simple name, and the type of each parameter
+   * @throws IOException if the attempt to read class path resources (jar files or directories)
+   * failed.
+   */
+  private static JsonObject extractMainNoticesProperties(String validatorPackageName)
+      throws IOException {
+    JsonObject toReturn = new JsonObject();
+    for (ClassPath.ClassInfo validatorClass : ClassPath.from(ClassLoader.getSystemClassLoader())
+            .getTopLevelClasses(validatorPackageName)) {
+      for (Class<?> innerNoticeClass : validatorClass.load().getDeclaredClasses()) {
+        if (ValidationNotice.class.isAssignableFrom(innerNoticeClass))
+        toReturn.add(
+            innerNoticeClass.getSimpleName(),
+            extractNoticeProperties(innerNoticeClass)
+        );
+      }
+    }
+    return toReturn;
+  }
+
+  /**
+   * Return a {@code JsonArray} that contains information about the type of each parameter of
+   * a {@code ValidationNotice} using the constructor of the class that is annotated by
+   * {@code  NoticeExport}.
+   *
+   * @param validationNoticeSubClass the {@code ValidationNotice} sub class to extract information
+   *                                from
+   * @return a {@code JsonArray} that contains information about the type of each parameter of
+   * said {@code ValidationNotice} using the constructor of the class that is annotated by
+   * {@code NoticeExport}.
+   */
+  private static JsonArray extractNoticeProperties(Class<?> validationNoticeSubClass) {
+    Constructor<?> constructor = getAnnotatedConstructor(validationNoticeSubClass);
+    JsonArray parametersAsJsonArray = new JsonArray();
+    Arrays.stream(constructor.getParameters()).forEach(parameter -> {
+      JsonObject parameterDetails = new JsonObject();
+      parameterDetails.addProperty(JSON_KEY_NAME, parameter.getName());
+        parameterDetails.addProperty(JSON_KEY_TYPE, mapDataType(parameter));
+      parametersAsJsonArray.add(parameterDetails);
+    });
+    if (!hasSeverityLevelParameter(constructor)) {
+      JsonObject severityDetails = new JsonObject();
+      severityDetails.addProperty(JSON_KEY_NAME, CLASS_SIMPLE_NAME_SEVERITY_LEVEL);
+      severityDetails.addProperty(JSON_KEY_TYPE, BLOB_DATA_TYPE);
```

Comment:
A severityLevel is a string from a fixed list (WARNING, ERROR, INFO). Why are you calling it BLOB?

Reply:
It's true, I was considering the type of the parameter passed to the Notice constructor instead of the type of the element that ends up in the final notice report. Modifying this.